pkg/services/folder/folderimpl/folder.go

```diff
@@ -319,6 +458,57 @@ func (s *Service) setFullpath(ctx context.Context, f *folder.Folder, user identi
 }
 
 func (s *Service) GetChildren(ctx context.Context, q *folder.GetChildrenQuery) ([]*folder.Folder, error) {
+	if s.features.IsEnabledGlobally(featuremgmt.FlagKubernetesFoldersServiceV2) {
+		return s.getChildrenFromApiServer(ctx, q)
+	}
+
+	return s.GetChildrenLegacy(ctx, q)
+}
+
+func (s *Service) getChildrenFromApiServer(ctx context.Context, q *folder.GetChildrenQuery) ([]*folder.Folder, error) {
+	// create a new context - prevents issues when the request stems from the k8s api itself
+	// otherwise the context goes through the handlers twice and causes issues
+	newCtx, cancel, err := s.getK8sContext(ctx)
+	if err != nil {
+		return nil, err
+	} else if cancel != nil {
+		defer cancel()
+	}
+
+	client, ok := s.k8sclient.getClient(newCtx, q.OrgID)
+	if !ok {
+		return nil, nil
+	}
+
+	out, err := client.List(newCtx, v1.ListOptions{})
```

Comment:
I'm actually thinking that for the folder tree implementation, I don't need a specific SQL template and it's maybe good enough to just list all folders within a namespace. So, I'm thinking on using a very similar approach as this one with just the namespace passed to ListOptions. The generic SQL template already allows for namespace filtering: https://github.com/grafana/grafana/blob/main/pkg/storage/unified/sql/data/resource_list.sql#L11